components/use-invoice.js

```diff
@@ -53,5 +53,5 @@ export default function useInvoice () {
     return newInvoice
   }, [retryPaidAction])
 
-  return { cancel, retry, isInvoice }
+  return useMemo(() => ({ cancel, retry, isInvoice }), [cancel, retry, isInvoice])
```

Comment:
This `useMemo` and the one in `useSendWallets` is pretty important since this avoid that `waitForWalletPayment` changes between renders which would mean that we run `useEffect` with the same failed invoices multiple times.

In general, this means it is very important that `waitForWalletPayment` is stable across renders.

Reply:
no longer the case since I am using  `useLazyQuery` + `setTimeout` now instead of relying on `useEffect`